### DISCUSSION TITLE
Initial distribution of HMM in simData

### DISCUSSION THREAD

**Discussion by anylital** [bmcclintock/momentuHMM#85] [Q&A] [answered]
Hi Brett, 

I'm trying to use simData function to produce realizations from a fitted HMM model. I want the state sequence to start from a certain state (say, state 1) and I thought I could do it by fixing the parameter delta. However, it seems that the first state is not sampled from the initial state distribution (delta) as I though. Here is my toy example with 2-state HMM, where I wanted the sequence to start from state 1:
 ```
dist <- list(step="gamma",angle="vm")
Par <- list(step=c(100,1000,50,100),angle=c(0,0,0.01,0.75))
set.seed(123)
data.sim<-simData(nbAnimals=2,obsPerAnimal=10,nbStates=2,dist=dist,Par=Par,
                    delta=c(1,0), states=TRUE)    
head(data.sim)
  ID       step      angle        x         y states
1  1  833.60385         NA   0.0000    0.0000      2
2  1 1041.50736 -0.1790091 341.4132  760.4817      2
3  1  951.04189  1.6199491 930.3381 1619.4957      2
4  1  115.06607 -1.8569732 120.4632 2118.0770      1
5  1 1128.34326 -2.0536619 205.9932 2195.0499      2
6  1   54.65923 -0.4443021 485.0652 1101.7625      1
 ```
So the first state seems not to honor the given initial distribution, because the first sequence starts from state 2 instead of state 1. 

A minor note: ```delta=c(0,1)``` in the previous example gives and error
 ```Error in sample.int(length(x), size, replace, prob) : NA in probability **vector.**```

The example given in the documentation of simData function seems to have the same issue, since one can get a realization where female begins in state 2, see below. (Note that I added "states=TRUE" to the original example.)  
```
# 9. Initial state (delta) based on covariate
nObs <- 100
dist <- list(step="gamma",angle="vm")
Par <- list(step=c(100,1000,50,100),angle=c(0,0,0.01,0.75))

# create sex covariate
cov <- data.frame(sex=factor(rep(c("F","M"),each=nObs))) # sex covariate
formulaDelta <- ~ sex + 0

# Female begins in state 1, male begins in state 2
delta <- matrix(c(-100,100),2,1,dimnames=list(c("sexF","sexM"),"state 2")) 

set.seed(123)
data.delta<-simData(nbAnimals=2,obsPerAnimal=nObs,nbStates=2,dist=dist,Par=Par,
                    delta=delta,formulaDelta=formulaDelta,covs=cov, states=TRUE)    
head(data.delta)
  ID       step      angle        x         y sex states
1  1  833.60385         NA   0.0000    0.0000   F      2
2  1 1041.50736 -0.1790091 341.4132  760.4817   F      2
3  1  951.04189  1.6199491 930.3381 1619.4957   F      2
4  1  115.06607 -1.8569732 120.4632 2118.0770   F      1
5  1 1128.34326 -2.0536619 205.9932 2195.0499   F      2
6  1   54.65923 -0.4443021 485.0652 1101.7625   F      1
```
Sampling of the first state seems to happen on line 1043: 
 ```Z[1] <- sample(1:nbStates,size=1,prob=delta0%*%gamma) ```

What is confusing me is that why this initial distribution delta is multiplied with gamma when sampling the first state? What am I missing here? And is there any way to guarantee that the simulated sequence starts from a certain state? 

Thanks in advance,
Anna-Kaisa

**Comment by bmcclintock** [comment, marked as answer]
Hi Anna-Kaisa,

Thank you for your interest in momentuHMM and posting your question here.  I assure you that the initial state is state 1 when delta = c(1,0).  What's happening in the example you provide above is that the tpm (beta on working scale; gamma on natural scale) is not specified and is therefore randomly drawn by simData.  The seed you set results in a relatively high probability of switching from 1 -> 2 and thus the state at time t=1 switches from that at time t=0 (where t=0 corresponds to the initial distribution).  See Eq. 1 in the vignette.  Note that the state at time t=0 is not returned when states=TRUE.

You're getting an error when delta=c(0,1) because parameters cannot be set on a boundary.  It squeaks by when delta=c(1,0) only because state 1 is the reference state on the working scale. The solution is to set delta=c(1.e-100, 1 - e.-100) or similar so that the parameters are not exactly on the boundary.  I'll add an error trap for this in the next release of momentuHMM (v1.5.0).

Cheers,
Brett

**Comment by anylital** [comment]
Thank you for the clarification. I'm still wondering: why is then the first row of my example (that should correspond t=1) showing that the sequence is at the initial location (0,0)? It indicates that there is no step (in space) between t=0 and t=1, even though the state changes.  

Cheers, 
Anna-Kaisa

**Comment by bmcclintock** [comment]
If it helps, you can instead think of the initial distribution (delta^(0)) in momentuHMM as the initial state probabilities for the time step from time t = -1 to time t = 0.  The state for the first time step (the first row in the data) corresponds to the interval from time t = 0 to time t = 1, and is drawn in simData with probability  \delta^(0) \Gamma^(1), where \Gamma^(1) is the state transition probability from the state for the time step from time t = -1 to time t = 0 to the state for the time step from time t = 0 to time t = 1.  

The step length for the first time step (the first row in the data) corresponds to the distance traveled from the initial location at time t = 0 (x and y in first row) to the location at time t = 1 (x and y in second row). Because step length and turn angle correspond to time intervals, there will always be T+1 rows in the location data but only T step lengths and T-1 turn angles.  

It's the same in moveHMM, the only difference in momentuHMM is that the initial distribution is multiplied by the t.p.m. for the first time step as in Eq. 1 of the vignette (i.e. \delta^(0) \Gamma^(1) P(z_1) \Gamma^(2) P(z_2)... instead of \delta^(1) P(z_1) \Gamma^(2) P(z_2)...). Thus \delta^(1) in moveHMM corresponds to \delta^(0) \Gamma^(1) in momentuHMM.  The momentuHMM parameterization allows covariates for the first observation (corresponding to time t=0) to be included in the t.p.m. for the first time step (\Gamma^(1)); moveHMM does not include covariates for time t=0.

**Comment by anylital** [comment]
Thanks again for the reply! I think I got it now. The last sentence clarified the reason for multiplying the initial distribution by the transition probability matrix, which was my main issue.
